app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -92,28 +134,41 @@ class ExplorationActivityPresenter @Inject constructor(
 
   fun showAudioStreamingOff() = exploreViewModel.isAudioStreamingOn.set(false)
 
-  fun setAudioBarVisibility(isVisible: Boolean) = getExplorationFragment()?.setAudioBarVisibility(isVisible)
+  fun setAudioBarVisibility(isVisible: Boolean) =
+    getExplorationFragment()?.setAudioBarVisibility(isVisible)
 
   fun scrollToTop() = getExplorationFragment()?.scrollToTop()
 
+  private fun getExplorationManagerFragment(): ExplorationManagerFragment? {
+    return activity.supportFragmentManager.findFragmentByTag(
+      TAG_EXPLORATION_MANAGER_FRAGMENT
+    ) as ExplorationManagerFragment?
+  }
+
   private fun getExplorationFragment(): ExplorationFragment? {
     return activity.supportFragmentManager.findFragmentById(
       R.id.exploration_fragment_placeholder
     ) as ExplorationFragment?
   }
 
   fun stopExploration() {
-    explorationDataController.stopPlayingExploration().observe(activity, Observer<AsyncResult<Any?>> {
-      when {
-        it.isPending() -> logger.d("ExplorationActivity", "Stopping exploration")
-        it.isFailure() -> logger.e("ExplorationActivity", "Failed to stop exploration", it.getErrorOrNull()!!)
-        else -> {
-          logger.d("ExplorationActivity", "Successfully stopped exploration")
-          backPressActivitySelector(backflowScreen)
-          (activity as ExplorationActivity).finish()
+    FontScaleConfigurationUtil(activity, StoryTextSize.MEDIUM_TEXT_SIZE.name).adjustFontScale()
```

Comment:
Isn't this happening in the manager fragment, instead?

Reply:
When i tried calling it in manager fragment it is updating 
UI late.